### PR TITLE
GUI: Work around Shiboken 6 bug for CallTips

### PR DESCRIPTION
**Pull request by pieterhijma**
See #15236, thanks @Jookia!

Shiboken 6.4.0 to 6.8.0 and newer crash when using getAttr on their objects. On these versions and newer, use a wrapper function to call getattr.

Fixes https://github.com/FreeCAD/FreeCAD/issues/14101
Obsoletes PR https://github.com/FreeCAD/FreeCAD/pull/14864
Obsoletes PR #15236

### CODE REVIEW

**Comment by pieterhijma** [comment]
@xtemp09, I've made a very small change in my commit to improve something that confused me a bit. I'm pretty sure it is correct, but feel free to test again.

**Comment by xtemp09** [comment]
@pieterhijma, I've tested your PR, it fixes the crash.


```ruby
OS: Ubuntu 22.04.4 LTS (KDE/plasma)
Word size of FreeCAD: 64-bit
Version: 0.22.0dev.38045 (Git)
Build type: Debug
Branch: RFC_shiboken6fix2
Hash: de3eb7578f73c90447ff1ba878077a805f9a8a0e
Python 3.10.12, Qt 6.7.1, Coin 4.0.0, Vtk , OCC 7.5.1
Locale: English/United States (en_US)
```


----




If I type `quit()` in Python console, _FreeCAD_ compiled with _Qt 6_ crashes, while _FreeCAD_ compiled with _Qt 5_ does not.


Here is the backtrace.

<details><summary>Backtrace</summary>


```cpp
Process 18351 stopped
* thread #1, name = 'FreeCAD', stop reason = signal SIGABRT
    frame #0: 0x00007ffff2e969fc libc.so.6`__GI___pthread_kill [inlined] __pthread_kill_implementation(no_tid=0, signo=6, threadid=140737238932480) at pthread_kill.c:44:76
(lldb) bt
* thread #1, name = 'FreeCAD', stop reason = signal SIGABRT
  * frame #0: 0x00007ffff2e969fc libc.so.6`__GI___pthread_kill [inlined] __pthread_kill_implementation(no_tid=0, signo=6, threadid=140737238932480) at pthread_kill.c:44:76
    frame #1: 0x00007ffff2e969b0 libc.so.6`__GI___pthread_kill [inlined] __pthread_kill_internal(signo=6, threadid=140737238932480) at pthread_kill.c:78:10
    frame #2: 0x00007ffff2e969b0 libc.so.6`__GI___pthread_kill(threadid=140737238932480, signo=6) at pthread_kill.c:89:10
    frame #3: 0x00007ffff2e42476 libc.so.6`__GI_raise(sig=6) at raise.c:26:13
    frame #4: 0x00007ffff2e287f3 libc.so.6`__GI_abort at abort.c:79:7
    frame #5: 0x00007ffff7894a86 libFreeCADGui.so`messageHandler(type=QtFatalMsg, context=<unavailable>, msg=0x00007fffffffc650) at Application.cpp:1823:14
    frame #6: 0x00007ffff3702aa7 libQt6Core.so.6`qt_message(QtMsgType, QMessageLogContext const&, char const*, __va_list_tag*) + 183
    frame #7: 0x00007ffff36ba1df libQt6Core.so.6`QMessageLogger::fatal(char const*, ...) const + 165
    frame #8: 0x00007ffff36b935c libQt6Core.so.6`qt_assert_x(char const*, char const*, char const*, int) + 96
    frame #9: 0x00007ffff7a9515a libFreeCADGui.so`void QtPrivate::assertObjectType<Gui::MainWindow>(o=0x00007fffffffd680) at qobjectdefs_impl.h:129:9
    frame #10: 0x00007ffff7d8e1b0 libFreeCADGui.so`QtPrivate::FunctorCall<QtPrivate::IndexesList<0>, QtPrivate::List<QMdiSubWindow*>, void, void (Gui::MainWindow::*)(QMdiSubWindow*)>::call(f=(libFreeCADGui.so`Gui::MainWindow::onWindowActivated(QMdiSubWindow*) at MainWindow.cpp:1317:1), o=0x00007fffffffd680, arg=0x00007fffffffc930) at qobjectdefs_impl.h:144:34
    frame #11: 0x00007ffff7d8e212 libFreeCADGui.so`QtPrivate::QCallableObject<void (Gui::MainWindow::*)(QMdiSubWindow*), QtPrivate::List<QMdiSubWindow*>, void>::impl(int, QtPrivate::QSlotObjectBase*, QObject*, void**, bool*) [inlined] void QtPrivate::FunctionPointer<void (Gui::MainWindow::*)(QMdiSubWindow*)>::call<QtPrivate::List<QMdiSubWindow*>, void>(arg=<unavailable>, o=<unavailable>, f=<unavailable>) at qobjectdefs_impl.h:182:95
    frame #12: 0x00007ffff7d8e20d libFreeCADGui.so`QtPrivate::QCallableObject<void (Gui::MainWindow::*)(QMdiSubWindow*), QtPrivate::List<QMdiSubWindow*>, void>::impl(which=<unavailable>, this_=<unavailable>, r=<unavailable>, a=<unavailable>, ret=<unavailable>) at qobjectdefs_impl.h:553:53
    frame #13: 0x00007ffff37def50 libQt6Core.so.6`void doActivate<false>(QObject*, int, void**) + 1744
    frame #14: 0x00007ffff4d3f7c3 libQt6Widgets.so.6`QMdiArea::subWindowActivated(QMdiSubWindow*) + 67
    frame #15: 0x00007ffff4d45dcb libQt6Widgets.so.6`QMdiAreaPrivate::_q_processWindowStateChanged(QFlags<Qt::WindowState>, QFlags<Qt::WindowState>) + 299
    frame #16: 0x00007ffff37df1df libQt6Core.so.6`void doActivate<false>(QObject*, int, void**) + 2399
    frame #17: 0x00007ffff4d4be59 libQt6Widgets.so.6`QMdiSubWindow::windowStateChanged(QFlags<Qt::WindowState>, QFlags<Qt::WindowState>) + 89
    frame #18: 0x00007ffff4d55e00 libQt6Widgets.so.6`QMdiSubWindowPrivate::setActive(bool, bool) + 752
    frame #19: 0x00007ffff4d480d2 libQt6Widgets.so.6`QMdiArea::event(QEvent*) + 114
    frame #20: 0x00007ffff4b8b345 libQt6Widgets.so.6`QApplicationPrivate::notify_helper(QObject*, QEvent*) + 133
    frame #21: 0x00007ffff7948254 libFreeCADGui.so`Gui::GUIApplication::notify(this=0x00007fffffffd6b0, receiver=0x0000555555b317b0, event=0x00007fffffffcee0) at GuiApplication.cpp:83:40
    frame #22: 0x00007ffff377b398 libQt6Core.so.6`QCoreApplication::notifyInternal2(QObject*, QEvent*) + 344
    frame #23: 0x00007ffff4be1150 libQt6Widgets.so.6`QWidgetPrivate::hideChildren(bool) + 416
    frame #24: 0x00007ffff4be1757 libQt6Widgets.so.6`QWidgetPrivate::hide_helper() + 327
    frame #25: 0x00007ffff4be5fda libQt6Widgets.so.6`QWidgetPrivate::setVisible(bool) + 922
    frame #26: 0x00007ffff4be1983 libQt6Widgets.so.6`QWidgetPrivate::handleClose(QWidgetPrivate::CloseMode) + 259
    frame #27: 0x00007ffff4bf789f libQt6Widgets.so.6`QWidgetWindow::closeEvent(QCloseEvent*) + 63
    frame #28: 0x00007ffff402d152 libQt6Gui.so.6`QWindow::event(QEvent*) + 1618
    frame #29: 0x00007ffff4b8b345 libQt6Widgets.so.6`QApplicationPrivate::notify_helper(QObject*, QEvent*) + 133
    frame #30: 0x00007ffff7948254 libFreeCADGui.so`Gui::GUIApplication::notify(this=0x00007fffffffd6b0, receiver=0x000055555ba822d0, event=0x00007fffffffd500) at GuiApplication.cpp:83:40
    frame #31: 0x00007ffff377b398 libQt6Core.so.6`QCoreApplication::notifyInternal2(QObject*, QEvent*) + 344
    frame #32: 0x00007ffff3fcce81 libQt6Gui.so.6`QGuiApplicationPrivate::processCloseEvent(QWindowSystemInterfacePrivate::CloseEvent*) + 177
    frame #33: 0x00007ffff403b38c libQt6Gui.so.6`bool QWindowSystemInterface::handleCloseEvent<QWindowSystemInterface::SynchronousDelivery>(QWindow*) + 204
    frame #34: 0x00007ffff4026bf0 libQt6Gui.so.6`QWindow::close() + 96
    frame #35: 0x00007ffff4be204e libQt6Widgets.so.6`QWidget::~QWidget() + 1150
    frame #36: 0x00007ffff7d84214 libFreeCADGui.so`Gui::MainWindow::~MainWindow(this=0x00007fffffffd680) at MainWindow.cpp:530:1
    frame #37: 0x00007ffff78983f5 libFreeCADGui.so`Gui::Application::runApplication() at Application.cpp:2163:1
    frame #38: 0x0000555555570ab7 FreeCAD`main(argc=<unavailable>, argv=<unavailable>) at MainGui.cpp:300:45
    frame #39: 0x00007ffff2e29d90 libc.so.6`__libc_start_call_main(main=(FreeCAD`main at MainGui.cpp:107:1), argc=1, argv=0x00007fffffffdd48) at libc_start_call_main.h:58:16
    frame #40: 0x00007ffff2e29e40 libc.so.6`__libc_start_main_impl(main=(FreeCAD`main at MainGui.cpp:107:1), argc=1, argv=0x00007fffffffdd48, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffdd38) at libc-start.c:392:3
    frame #41: 0x00005555555702e5 FreeCAD`_start + 37
(lldb)
```

</details>

**Comment by chennes** [comment]
@xtemp09 is that a new bug you are reporting, or a continuation of this one?

**Comment by xtemp09** [comment]
@chennes, so far this is just information for pieterhijma; it looks like another _Qt6_-related bug.

**Comment by pieterhijma** [comment]
I can confirm with a similar stack trace. It doesn't happen when all MDI views are closed. When a file is open or the Start workbench is in its MDI view, I get the stacktrace below.

Since it is not related to the code from this PR, it should be filed as a new issue. I can do it, but @xtemp09, because you found it first, the honor is all yours :slightly_smiling_face:.

```
#0  __pthread_kill_implementation (threadid=<optimized out>, signo=signo@entry=6, no_tid=no_tid@entry=0) at pthread_kill.c:44
#1  0x00007ffff04a8eb3 in __pthread_kill_internal (threadid=<optimized out>, signo=6) at pthread_kill.c:78
#2  0x00007ffff0450a30 in __GI_raise (sig=sig@entry=6) at ../sysdeps/posix/raise.c:26
#3  0x00007ffff04384c3 in __GI_abort () at abort.c:79
#4  0x00007ffff679e93f in messageHandler (type=QtFatalMsg, context=..., msg=...) at /home/pieter/current/ondsel/freecad-fix-shiboken6/src/Gui/Application.cpp:1823
#5  0x00007ffff0adcf24 in qt_message_print (msgType=QtFatalMsg, context=..., message=...) at /usr/src/debug/qt6-base/qtbase/src/corelib/global/qlogging.cpp:2018
#6  0x00007ffff0add07e in qt_message(QtMsgType, const QMessageLogContext &, const char *, typedef __va_list_tag __va_list_tag *) (msgType=msgType@entry=QtFatalMsg, context=..., msg=msg@entry=0x7ffff0e21fb8 "ASSERT failure in %s: \"%s\", file %s, line %d", ap=ap@entry=0x7fffffffbcf0) at /usr/src/debug/qt6-base/qtbase/src/corelib/global/qlogging.cpp:371
#7  0x00007ffff0a8cde7 in QMessageLogger::fatal (this=this@entry=0x7fffffffbdd8, msg=msg@entry=0x7ffff0e21fb8 "ASSERT failure in %s: \"%s\", file %s, line %d") at /usr/src/debug/qt6-base/qtbase/src/corelib/global/qlogging.cpp:889
#8  0x00007ffff0a8ce98 in qt_assert_x (where=<optimized out>, what=<optimized out>, file=<optimized out>, line=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/corelib/global/qassert.cpp:77
#9  0x00007ffff6ba39ea in QtPrivate::assertObjectType<Gui::MainWindow> (o=0x7fffffffcdc0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:129
#10 0x00007ffff70bc47f in QtPrivate::FunctorCall<QtPrivate::IndexesList<0>, QtPrivate::List<QMdiSubWindow*>, void, void (Gui::MainWindow::*)(QMdiSubWindow*)>::call (f=(void (Gui::MainWindow::*)(class Gui::MainWindow * const, class QMdiSubWindow *)) 0x7ffff709fc3a <Gui::MainWindow::onWindowActivated(QMdiSubWindow*)>, o=0x7fffffffcdc0, arg=0x7fffffffbff0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:144
#11 0x00007ffff70b8788 in QtPrivate::FunctionPointer<void (Gui::MainWindow::*)(QMdiSubWindow*)>::call<QtPrivate::List<QMdiSubWindow*>, void> (f=(void (Gui::MainWindow::*)(class Gui::MainWindow * const, class QMdiSubWindow *)) 0x7ffff709fc3a <Gui::MainWindow::onWindowActivated(QMdiSubWindow*)>, o=0x7fffffffcdc0, arg=0x7fffffffbff0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:182
#12 0x00007ffff70b54a7 in QtPrivate::QCallableObject<void (Gui::MainWindow::*)(QMdiSubWindow*), QtPrivate::List<QMdiSubWindow*>, void>::impl (which=1, this_=0x555555d83aa0, r=0x7fffffffcdc0, a=0x7fffffffbff0, ret=0x0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:553
#13 0x00007ffff0ba16f7 in QtPrivate::QSlotObjectBase::call (this=<optimized out>, r=<optimized out>, a=<optimized out>, this=<optimized out>, r=<optimized out>, a=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/corelib/kernel/qobjectdefs_impl.h:469
#14 doActivate<false> (sender=<optimized out>, signal_index=<optimized out>, argv=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/corelib/kernel/qobject.cpp:4086
#15 0x00007ffff1cb62d9 in QMdiArea::subWindowActivated (this=0x555555c3c240, _t1=0x0) at /usr/src/debug/qt6-base/build/src/widgets/Widgets_autogen/include/moc_qmdiarea.cpp:346
#16 QMdiAreaPrivate::resetActiveWindow (this=0x55555590ae60, deactivatedWindow=0x55555959da00) at /usr/src/debug/qt6-base/qtbase/src/widgets/widgets/qmdiarea.cpp:1060
#17 QMdiAreaPrivate::_q_processWindowStateChanged (this=0x55555590ae60, oldState=..., newState=...) at /usr/src/debug/qt6-base/qtbase/src/widgets/widgets/qmdiarea.cpp:718
#18 QMdiArea::qt_static_metacall (_o=<optimized out>, _c=<optimized out>, _id=<optimized out>, _a=<optimized out>) at /usr/src/debug/qt6-base/build/src/widgets/Widgets_autogen/include/moc_qmdiarea.cpp:246
#19 0x00007ffff0ba194a in doActivate<false> (sender=<optimized out>, signal_index=<optimized out>, argv=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/corelib/kernel/qobject.cpp:4098
#20 0x00007ffff1cc025c in QMdiSubWindow::windowStateChanged (this=<optimized out>, _t1=..., _t2=...) at /usr/src/debug/qt6-base/build/src/widgets/Widgets_autogen/include/moc_qmdisubwindow.cpp:241
#21 QMdiSubWindowPrivate::setActive (this=0x5555593f1360, activate=<optimized out>, changeFocus=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/widgets/widgets/qmdisubwindow.cpp:1450
#22 0x00007ffff1cb4cc5 in QMdiAreaPrivate::setActive (active=false, changeFocus=false, this=<optimized out>, subWindow=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/widgets/widgets/qmdiarea_p.h:219
#23 QMdiAreaPrivate::setActive (this=0x55555590ae60, subWindow=<optimized out>, active=false, changeFocus=false) at /usr/src/debug/qt6-base/qtbase/src/widgets/widgets/qmdiarea_p.h:219
#24 QMdiArea::event (this=0x555555c3c240, event=0x7fffffffc570) at /usr/src/debug/qt6-base/qtbase/src/widgets/widgets/qmdiarea.cpp:2528
#25 0x00007ffff1afc51c in QApplicationPrivate::notify_helper (this=<optimized out>, receiver=0x555555c3c240, e=0x7fffffffc570) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qapplication.cpp:3287
#26 0x00007ffff6947e7a in Gui::GUIApplication::notify (this=0x7fffffffcdf0, receiver=0x555555c3c240, event=0x7fffffffc570) at /home/pieter/current/ondsel/freecad-fix-shiboken6/src/Gui/GuiApplication.cpp:83
#27 0x00007ffff0b44d28 in QCoreApplication::notifyInternal2 (receiver=0x555555c3c240, event=0x7fffffffc570) at /usr/src/debug/qt6-base/qtbase/src/corelib/kernel/qcoreapplication.cpp:1142
#28 0x00007ffff0b44d6d in QCoreApplication::sendEvent (receiver=<optimized out>, event=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/corelib/kernel/qcoreapplication.cpp:1583
#29 0x00007ffff1b459e1 in QWidgetPrivate::hideChildren (this=this@entry=0x555555bde560, spontaneous=spontaneous@entry=false) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qwidget.cpp:8556
#30 0x00007ffff1b45f92 in QWidgetPrivate::hide_helper (this=this@entry=0x555555bde560) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qwidget.cpp:8271
#31 0x00007ffff1b49a9a in QWidgetPrivate::setVisible (this=0x555555bde560, visible=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qwidget.cpp:8463
#32 0x00007ffff1b4681a in QWidget::hide (this=0x7fffffffcdc0) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qwidget.cpp:8240
#33 QWidgetPrivate::handleClose (this=0x555555bde560, mode=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qwidget.cpp:8630
#34 0x00007ffff1b63067 in QWidgetWindow::closeEvent (this=<optimized out>, event=0x7fffffffcbf0) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qwidgetwindow.cpp:832
#35 0x00007ffff11df4e0 in QWindow::event (this=0x5555582b3290, ev=0x7fffffffcbf0) at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qwindow.cpp:2587
#36 0x00007ffff1afc51c in QApplicationPrivate::notify_helper (this=<optimized out>, receiver=0x5555582b3290, e=0x7fffffffcbf0) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qapplication.cpp:3287
#37 0x00007ffff6947e7a in Gui::GUIApplication::notify (this=0x7fffffffcdf0, receiver=0x5555582b3290, event=0x7fffffffcbf0) at /home/pieter/current/ondsel/freecad-fix-shiboken6/src/Gui/GuiApplication.cpp:83
#38 0x00007ffff0b44d28 in QCoreApplication::notifyInternal2 (receiver=0x5555582b3290, event=0x7fffffffcbf0) at /usr/src/debug/qt6-base/qtbase/src/corelib/kernel/qcoreapplication.cpp:1142
#39 0x00007ffff0b44d7d in QCoreApplication::sendSpontaneousEvent (receiver=<optimized out>, event=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/corelib/kernel/qcoreapplication.cpp:1597
#40 0x00007ffff1174f94 in QGuiApplicationPrivate::processCloseEvent (e=0x7fffffffcc50) at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qguiapplication.cpp:2745
#41 QGuiApplicationPrivate::processCloseEvent (e=0x7fffffffcc50) at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qguiapplication.cpp:2733
#42 0x00007ffff11bc7bb in QWindowSystemEventHandler::sendEvent (this=<optimized out>, e=0x7fffffffcc50) at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qwindowsysteminterface.cpp:190
#43 QWindowSystemHelper<QWindowSystemInterface::SynchronousDelivery>::handleEvent<QWindowSystemInterfacePrivate::CloseEvent, QWindow*> () at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qwindowsysteminterface.cpp:102
#44 QWindowSystemHelper<QWindowSystemInterface::SynchronousDelivery>::handleEvent<QWindowSystemInterfacePrivate::CloseEvent, QWindow*> () at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qwindowsysteminterface.cpp:96
#45 handleWindowSystemEvent<QWindowSystemInterfacePrivate::CloseEvent, QWindowSystemInterface::SynchronousDelivery, QWindow*> () at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qwindowsysteminterface.cpp:138
#46 QWindowSystemInterface::handleCloseEvent<QWindowSystemInterface::SynchronousDelivery> (window=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qwindowsysteminterface.cpp:351
#47 QPlatformWindow::close (this=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qplatformwindow.cpp:329
#48 0x00007ffff11d78a7 in QWindow::close (this=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/gui/kernel/qwindow.cpp:2371
#49 0x00007ffff1b33816 in QWidget::~QWidget (this=<optimized out>, this=<optimized out>) at /usr/src/debug/qt6-base/qtbase/src/widgets/kernel/qwidget.cpp:1510
#50 0x00007ffff709af61 in Gui::MainWindow::~MainWindow (this=0x7fffffffcdc0, __in_chrg=<optimized out>) at /home/pieter/current/ondsel/freecad-fix-shiboken6/src/Gui/MainWindow.cpp:530
#51 0x00007ffff679ff98 in Gui::Application::runApplication () at /home/pieter/current/ondsel/freecad-fix-shiboken6/src/Gui/Application.cpp:2163
#52 0x000055555557a579 in main (argc=1, argv=0x7fffffffd5d8) at /home/pieter/current/ondsel/freecad-fix-shiboken6/src/Main/MainGui.cpp:300
```

File: src/Gui/CallTips.cpp [hunk 1]
```diff
@@ -340,24 +340,52 @@ QMap<QString, CallTip> CallTipsList::extractTips(const QString& context) const
     return tips;
 }
 
+bool shibokenMayCrash(void)
+{
+    // Shiboken 6.4.0 to 6.8.0 crash if we try to read their object
+    // atttributes without a current stack frame.
```
Comment (chennes) [inline src/Gui/CallTips.cpp:346]:
```suggestion
    // attributes without a current stack frame.
```

Comment (pieterhijma) [reply src/Gui/CallTips.cpp]: Thanks, fixed with an amend.

File: src/Gui/CallTips.cpp [hunk 2]
```diff
@@ -340,24 +340,52 @@ QMap<QString, CallTip> CallTipsList::extractTips(const QString& context) const
     return tips;
 }
 
+bool shibokenMayCrash(void)
+{
+    // Shiboken 6.4.0 to 6.8.0 crash if we try to read their object
```
Comment (chennes) [inline src/Gui/CallTips.cpp:345]: The version numbers here don't line up with the code below.

**Review by chennes — APPROVED** [review]
This works for me to fix the bug (tested on Win11 with Qt 6.7.2).

File: src/Gui/CallTips.cpp (hunk 2 above)
Comment (chennes) [inline src/Gui/CallTips.cpp:345]:
```suggestion
    // Shiboken 6.4.8 to 6.7.3 crash if we try to read their object
```

**Comment by chennes** [comment]
(If we don't hear from @pieterhijma today, I'll merge without the fixed comment)

**Pull request merged**